### PR TITLE
add statefulSet.go

### PR DESCRIPTION
**Pull request by bzsuni**

### CODE REVIEW

File: framework/statefulset.go [hunk 1]
```diff
@@ -1,3 +1,124 @@
...
+	key := client.ObjectKeyFromObject(sts)
+	existing := &appsv1.StatefulSet{}
+	e := f.GetResource(key, existing)
+	if e != nil {
+		return nil, e
+	}
+	return existing, e
+}
+
+func (f *Framework) WaitStsAllReplicaReady(name, namespace string, ctx context.Context) (*appsv1.StatefulSet, error) {
```
Comment (weizhoublue) [inline framework/statefulset.go:76]: 其它方法 都是 ****StatefulSet***， 维度此处简写，请统一规范 ， 
诸如  WaitStatefulSetReady  是否就好了，不过过多修饰

Comment (bzsuni) [reply framework/statefulset.go]: 方法名改好了，，

**Comment by weizhoublue** [comment]
注意解决 pr check 失败问题

File: framework/statefulSet_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		replica := int32(3)
+		readyReplica := int32(3)
+		sts := generateExampleStatefulSetYaml(stsName, namespace, replica, readyReplica)
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+
+		go func() {
+			_, err := f.WaitStatefulSetReady(stsName, namespace, ctx)
+			Expect(err).NotTo(HaveOccurred())
+		}()
```
Comment (weizhoublue) [inline framework/statefulSet_test.go:77]: 为什么 要启动新的协程，
而不是  顺序的 
f.CreateStatefulSet
f.WaitStatefulSetReady

Comment (weizhoublue) [reply framework/statefulSet_test.go:77]: goroutine 11 [running]:
[149](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:149)
  github.com/onsi/ginkgo/v2.Fail({0xc000646150, 0x69}, {0xc000364068, 0xc000646150, 0x0})
[150](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:150)
  	/home/runner/work/e2eframework/e2eframework/vendor/github.com/onsi/ginkgo/v2/core_dsl.go:335 +0x225
[151](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:151)
  github.com/onsi/gomega/internal.(*Assertion).match(0xc0000b4040, {0x178b718, 0x2374d18}, 0x0, {0x0, 0x0, 0x0})
[152](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:152)
  	/home/runner/work/e2eframework/e2eframework/vendor/github.com/onsi/gomega/internal/assertion.go:100 +0x1f0
[153](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:153)
  github.com/onsi/gomega/internal.(*Assertion).NotTo(0xc0000b4040, {0x178b718, 0x2374d18}, {0x0, 0x0, 0x0})
[154](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:154)
  	/home/runner/work/e2eframework/e2eframework/vendor/github.com/onsi/gomega/internal/assertion.go:68 +0x91
[155](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:155)
  github.com/spidernet-io/e2eframework/framework_test.glob..func4.2.1()
[156](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:156)
  	/home/runner/work/e2eframework/e2eframework/framework/statefulSet_test.go:76 +0x71
[157](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:157)
  created by github.com/spidernet-io/e2eframework/framework_test.glob..func4.2
[158](https://github.com/spidernet-io/e2eframework/runs/6438193566?check_suite_focus=true#step:5:158)
  	/home/runner/work/e2eframework/e2eframework/framework/statefulSet_test.go:74 +0x10f

Comment (bzsuni) [reply framework/statefulSet_test.go:77]: 顺序执行，捕捉不到事件，单元测试报错，不知道为什么
image
image

File: framework/framework.go [hunk 3]
```diff
@@ -196,6 +205,12 @@ func (f *Framework) ListResource(list client.ObjectList, opts ...client.ListOpti
 	return f.KClient.List(ctx3, list, opts...)
 }
 
+func (f *Framework) UpdateResource(obj client.Object, opts ...client.UpdateOption) error {
+	ctx4, cancel4 := context.WithTimeout(context.Background(), f.Config.ApiOperateTimeout)
```
Comment (weizhoublue) [inline framework/framework.go:209]: 该 api 没有 单元测试，如果不是 本次 要加的，而是 顺带加的，

建议要么不加，要么补足单元测试 ，避免成为 覆盖率死角，后续无人负责

因为我们希望本工程 是 严格 质量把控的，有多项目在共享

**Pull request closed without merging**